pkg/logs/json.go

```diff
@@ -0,0 +1,72 @@
+package logs
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+type JSONConfig struct {
+	Mapping map[string]string `yaml:"mapping"`
+}
+
+type JSONParser struct {
+	reader  *bufio.Reader
+	mapping map[string]string
+}
+
+func NewJSONParser(config JSONConfig, in io.Reader) (*JSONParser, error) {
+	fieldMap := config.Mapping
+	if fieldMap == nil {
+		fieldMap = make(map[string]string)
+	}
+
+	parser := &JSONParser{
+		reader:  bufio.NewReader(in),
+		mapping: fieldMap,
+	}
+
+	return parser, nil
+}
+
+func (p *JSONParser) ReadLine(line LogLine) error {
+	row, err := p.reader.ReadSlice('\n')
+
+	if err != nil && len(row) == 0 {
+		return err
+	}
+
+	return p.Parse(row, line)
+}
+
+func (p *JSONParser) Parse(row []byte, line LogLine) error {
+	var parsedLine map[string]interface{}
+
+	if err := json.Unmarshal(row, &parsedLine); err != nil {
+		return err
+	}
+
+	/* Now map the fields */
+	for logField, logValue := range parsedLine {
+		/* Convert logValue to string */
+		stringValue := fmt.Sprintf("%v", logValue)
```

Comment:
Why do we need this? Are there something we cant unmarshal into a `string`?

Reply:
I convert it back to string because `weblog/logline::Assign()` expects the `value` argument to be a string. The json log that is being parsed might store integers either as strings or, well, integers. In effect we're not sure of the type of the value, hence the cast to string. The same goes for floats and bools.